experiments/process_sandbox/README.md

```diff
@@ -330,3 +329,184 @@ Code layout
+  |                                                                                                                                            |-- 0x7fbe80000000 [r--s]
+  |                                                                                                                                            |
+  ...                                                                                           |-- 0x7fbf00000000 [r--s]                      |
+  |                                                                                             |                                              |
+  |                                              |-- 0x7fbf80000000 [r--s]                      |                                              |
+  |                                              |                                              |                                              |        PageMap
+  |-- 0x7fc000000000                             |                                              |                                              |
+                                                 |                                              |        PageMap                               |
+                                                 |                                              |                                              |
+                                                 |        PageMap                               |                                              | 
+  |-- 0x7ffe80000000 -- 0x7ffec0000000 [rw-s]    |                                              |                                              |-- child 3 region [rw-s]
+                                                 |                                              |                                            
+  |-- 0x7fff00000000 -- 0x7fff40000000 [rw-s]    |                                              |-- child 2 region [rw-s]                     
+                                                 |                                                                                           
+  |-- 0x7fff80000000 -- 0x7fffc0000000 [rw-s]    |-- child 1 region [rw-s]                                                                   
+
+```
+
+The `library_runner` is responsible, at initialization time, for ensuring that both the `PageMap` and the child's memory region (i.e., heap) are correctly mapped in the process's address space.
+For that purposes, it uses the `memfd` file descriptors inherited from the parent.
+It relies on `mmap` to install the heap as a shared memory region that is both readable and writable.
+For the `PageMap`, it resizes the region and drops write access to the corresponding file descriptor using `mmap`.
+Despite observable overlaps between the `PageMap` memory regions and heaps of different children, this initialization step ensures that the corresponding virtual memory is backed by separate physical pages in the different children address spaces. 
```

Comment:
I'm not sure what this means.  The key thing is that the same physical memory backs the pagemap, it doesn't matter where the pagemap is in the child's address space because it's accessed via a pointer.  This is an implementation detail that's subject to change: we may want to put the pagemap at the same place in all processes in the future (and we did in the pre-snmalloc2 version).